backend/router/validator.py

```
@@ -4470,24 +4472,1743 @@ def _topic_to_camera(self, topic: str) -> str:
+                after_hp = context["after"].get("hp", 0)
+
+            before_hp = self._safe_get_nested(context["before"], "hp.current", None)
+            if before_hp is None:
+                before_hp = context["before"].get("hp", 0)
+
+            # Replace values in condition string
+            eval_condition = condition.replace("after.hp.current", str(after_hp))
+            eval_condition = eval_condition.replace("before.hp.current", str(before_hp))
+
+            try:
+                return eval(eval_condition)
```

Ruff (0.13.3)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)